schematic/configuration.py

```diff
@@ -0,0 +1,112 @@
+import os
+import yaml
+
+
+class Configuration(object):
+
+    def __init__(self):
+        # this is the project's root directory
+        self.ROOT_DIR = None
+        # path to config.yml file
+        self.CONFIG_PATH = None
+        # path to credentials.json file
+        self.CREDS_PATH = None
+        # path to token.pickle file
+        self.TOKEN_PICKLE = None
+        # path to data directory
+        self.DATA_PATH = None
+        # path to service account credentials file
+        self.SERVICE_ACCT_CREDS = None
+        # path to synapse config file
+        self.SYNAPSE_CONFIG_PATH = None
+        # entire configuration data
+        self.DATA = None
+
+
+    def __getattribute__(self, name):
+        value = super().__getattribute__(name)
+        if value is None and "SCHEMATIC_CONFIG" in os.environ:
+            self.load_config_from_env()
+            value = super().__getattribute__(name)
+        elif value is None and "SCHEMATIC_CONFIG" not in os.environ:
+            raise AttributeError(
+                "The '%s' configuration field was accessed, but it hasn't been "
+                "set yet, presumably because the schematic.CONFIG.load_config() "
+                "method hasn't been run yet. Alternatively, you can re-run this "
+                "code with the 'SCHEMATIC_CONFIG' environment variable set to "
+                "the config.yml file, which will be automatically loaded."
+                % name
+            )
+        return value
+
+
+    def __getitem__(self, key):
```

Comment:
Just a small shortcut so users can use `CONFIG[key]` instead of having to know that the config dictionary is in `CONFIG.DATA`.